### PR TITLE
memtier: properly reconfigure/start/stop dynamic demoter.

### PR DESCRIPTION
**Pull request by klihub**
The commit to disable the dynamic demoter by default accidentally put it on a permanent vacation. It is not configured correctly for the initial configuration (unless that has it disabled) and since it does not get properly reconfigured upon configuration changes, this prevents it from ever being activated. This patch should fix that.

### CODE REVIEW

File: pkg/cri/resource-manager/policy/builtin/memtier/dynamic_demotion.go [hunk 1]
```diff
@@ -117,6 +114,28 @@ func copyPagePool(p pagePool) pagePool {
 	return c
 }
 
+func CreateDemoter(p *policy, pageMover PageMover) *demoter {
```
Comment (kad) [inline pkg/cri/resource-manager/policy/builtin/memtier/dynamic_demotion.go:117]: Minor nit: most of go libraries are using convention for Object, there is NewObject function. Maybe here also NewDemoter?

**Review by kad — COMMENTED** [review]
lgtm, with very small nit.

File: pkg/cri/resource-manager/policy/builtin/memtier/dynamic_demotion.go (hunk 1 above)
Comment (klihub) [reply pkg/cri/resource-manager/policy/builtin/memtier/dynamic_demotion.go:117]: Ok, I'll update.

**Review by marquiz — COMMENTED** [review]
Looks, good. The only comment/nit I have is that the exported functions/method (`NewDemoter`, Reconfigure`, are there others(?)) should have a "docstring" comment

**Pull request merged**